steps/generate_analytical_dataset.py

```diff
@@ -8,56 +8,90 @@
 import re
 import os
 import concurrent.futures
+import time
+import datetime
+import os
 
 from Crypto.Cipher import AES
 from Crypto.Util import Counter
 from pyspark.sql.types import *
 from pyspark.sql import Row
 from datetime import datetime
-import logging
 import pytz
+from logger import setup_logging
+
+
+the_logger = setup_logging(
+    log_level=os.getenv("LOG_LEVEL"),
```

Comment:
this will fail if env var is missing or not uppercase, so I suggest the following:
```suggestion
    log_level = os.environ["ADG_LOG_LEVEL"].upper() if "ADG_LOG_LEVEL" in os.environ else "INFO"
```